docs/mxdoc.py

```diff
@@ -109,7 +109,13 @@ def build_scala_docs(app):
     """build scala doc and then move the outdir"""
     scala_path = app.builder.srcdir + '/../scala-package'
     # scaldoc fails on some apis, so exit 0 to pass the check
-    _run_cmd('cd ' + scala_path + '; scaladoc `find . -type f -name "*.scala" | egrep \"\/core|\/infer\" | egrep -v \"Suite\"`; exit 0')
+    scala_doc_sources = 'find . -type f -name "*.scala" | egrep \"\.\/core|\.\/infer\" | egrep -v \"Suite\"'
+    scala_doc_classpath = ':'.join([
+        '`find native -name "*.jar" | grep "target/lib/" | tr "\n" ":" `',
+        '`find macros -name "*-SNAPSHOT.jar" | tr "\n" ":" `'
```

Comment:
Not sure why include macros SNAPSHOT?

Reply:
We only build the scaladocs for the core and infer modules. So, I am treating the macros build as a dependency (which is required to avoid some missing dependency errors). The other option is that we also build docs for the macros as well, but I think it is better to hide them since it is not really for the users